### PR TITLE
Implement Comment section with (buggy) submit and delete features

### PR DESCRIPTION
**Pull request by bradleypickard**
Contains a large number of changes to the Portfolio, but code relevant to the Comment section is located in index.html, script.js, DataServlet.java and DeleteDataServlet.java.

At the moment, submitting an deleting comments only sometimes causes the correct change to be reflected on-screen. However, within Datastore the correct Entities are being put() or delete(). As a result, simply refreshing the page will cause it to look how it should.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -23,10 +34,75 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<String> comments;
```
Comment (hanpeiz) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:37]: Is there a reason for this to be a class variable? Looks like it is essentially only used in doGet (and used like a "global" variable to pass a parameter to writeGetResponse()).

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+public class DeleteDataServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    System.out.println("Comment delete initiated.");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment");
+    PreparedQuery results = datastore.prepare(query);
+    ArrayList<Key> keys = new ArrayList<Key>();
+    for (Entity entity : results.asIterable()) {
+      keys.add(entity.getKey());
```
Comment (hanpeiz) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:45]: Can you just delete the key here in this loop here?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -23,10 +34,75 @@
...
+    Query query = new Query("Comment");
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("body");
+      comments.add(comment);
+    }
+    System.out.println("Finished Query. Size: " + comments.size());
+
+    String maxCommentParam = request.getParameter("max-comments");
+    int displayedComments = tryParseInt(maxCommentParam);
```
Comment (hanpeiz) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:59]: Yay server-side validation!

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -12,41 +12,110 @@
...
 
   // Add it to the page.
-  const container = document.getElementById('greeting-container');
-  container.innerText = greeting;
-  
+  const container = document.getElementById('avatar-container');
+  container.innerText = line;
 }
 
+function getComments() {  // eslint-disable-line no-unused-vars
```
Comment (hanpeiz) [inline portfolio/src/main/webapp/script.js:61]: Curious - why is this needed? I don't see any unused vars here.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (bradleypickard) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:37]: That's something I've been meaning to change. I used to store comments in that variable between calls to doGet to reduce my interactions with Datastore but it seemed like a bad practice because the servlet could get out of sync.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 2 above)
Comment (bradleypickard) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:45]: I could, and that's how I used to do it. The delete operation was separated out with the intention to atomize it but that hasn't seemed to help anyway.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (bradleypickard) [reply portfolio/src/main/webapp/script.js:61]: I added this before getComments() was called elsewhere in the script. I'll make sure to delete!
